rustls/src/enums.rs

```diff
@@ -544,15 +545,40 @@ impl SignatureScheme {
     ///
     /// See RFC8446 s4.2.3.
     pub(crate) fn supported_in_tls13(&self) -> bool {
-        matches!(
-            *self,
-            Self::ECDSA_NISTP521_SHA512
-                | Self::ECDSA_NISTP384_SHA384
-                | Self::ECDSA_NISTP256_SHA256
-                | Self::RSA_PSS_SHA512
-                | Self::RSA_PSS_SHA384
-                | Self::RSA_PSS_SHA256
-                | Self::ED25519
+        let [hash, sign] = self.to_array();
+
+        // This covers both disallowing SHA1:
+        //
+        //  "These values refer solely to signatures which appear in certificates
+        //   (see Section 4.4.2.2) and are not defined for use in signed TLS
+        //   handshake messages"
+        //
+        // and old hash functions:
+        //
+        //  "These algorithms are deprecated as of TLS 1.3.  They MUST NOT be
+        //   offered or negotiated by any implementation.  In particular, MD5
+        //   [SLOTH], SHA-224, and DSA MUST NOT be used."
+        match HashAlgorithm::from(hash) {
+            HashAlgorithm::NONE
+            | HashAlgorithm::MD5
+            | HashAlgorithm::SHA1
+            | HashAlgorithm::SHA224 => return false,
+            _ => (),
+        };
+
+        // RSA-PKCS1 is also disallowed for TLS1.3:
+        //
+        //  "These values refer solely to signatures which appear in certificates
+        //   (see Section 4.4.2.2) and are not defined for use in signed TLS
+        //   handshake messages"
```

Comment:
Does this repeat the same quote? Not sure if that's useful.

Reply:
And this one uncut is:

```
RSASSA-PKCS1-v1_5 algorithms:  Indicates a signature algorithm using
      RSASSA-PKCS1-v1_5 [RFC8017] with the corresponding hash algorithm
      as defined in [SHS].  These values refer solely to signatures
      which appear in certificates (see Section 4.4.2.2) and are not
      defined for use in signed TLS handshake messages, although they
      MAY appear in "signature_algorithms" and
      "signature_algorithms_cert" for backward compatibility with
      TLS 1.2.
```

Reply:
Going to replace these quotes with references to the paragraphs.